packages/api/src/external/fhir/document/draft-update-document-reference.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  Attachment,
+  DocumentReference,
+  DocumentReferenceContent,
+  Organization,
+} from "@medplum/fhirtypes";
+import { createFHIRSubject } from ".";
+import BadRequestError from "../../../errors/bad-request";
+import { OrganizationModel } from "../../../models/medical/organization";
+import { metriportDataSourceExtension } from "../shared/extensions/metriport";
+
+export function docRefCheck(docRef: DocumentReference) {
+  if (!docRef.description) throw new BadRequestError(`Document Reference must have a description`);
+  if (!docRef.type) throw new BadRequestError(`Document Reference must have a type`);
+  if (!docRef.context) throw new BadRequestError(`Document Reference must have context`);
+}
+
+export function pickDocRefParts(
+  prelimDocRef: DocumentReference,
+  organization: OrganizationModel,
+  docRefId: string,
+  patientId: string,
+  s3Key: string,
+  bucketName: string
+): DocumentReference {
+  const containedOrg: Organization = {
+    resourceType: "Organization",
+    id: organization.id,
+    identifier: [{ value: organization.dataValues.organizationNumber.toString() }],
+    name: organization.dataValues.data.name,
+  };
+  const temporaryExtension = metriportDataSourceExtension;
+  temporaryExtension.valueCoding.code = "TEMPORARY";
```

Comment:
I suggest we create an external const for this `TEMPORARY` literal. Even if not referenced externally now, its good to keep those separated from the logic for visibility and reuse.